src/vm/gcheaputilities.cpp

```diff
@@ -152,6 +152,69 @@ void StashKeywordAndLevel(bool isPublicProvider, GCEventKeyword keywords, GCEven
     }
 }
 
+#ifdef FEATURE_STANDALONE_GC
+HMODULE LoadStandaloneGc(LPCWSTR libFileName)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    //
+    // Look for the standalone GC module next to the hosting binary
+    //
+
+    PathString libPath;
+    DWORD result = WszGetModuleFileName(nullptr, libPath);
+    if (result == 0)
+    {
+        LOG((LF_GC, LL_FATALERROR, "GetModuleFileName failed, function 'LoadStandaloneGc': error %u\n", GetLastError()));
+        return nullptr;
+    }
+
+    assert(libFileName);
+    PathString::Iterator iter = libPath.End();
+    if (libPath.FindBack(iter, DIRECTORY_SEPARATOR_CHAR_W))
+    {
+        libPath.Truncate(++iter);
+        libPath.Append(libFileName);
+    }
+    else
+    {
+        assert(false && "unreachable");
+    }
+
+    LPCWSTR libraryName = libPath.GetUnicode();
+    LOG((LF_GC, LL_INFO100, "Loading standalone GC from path %S\n", libraryName));
+    HMODULE libraryHandle = CLRLoadLibrary(libraryName);
+    if (libraryHandle != nullptr)
+        return libraryHandle;
+
+    LOG((LF_GC, LL_INFO100, "Failed to load standalone GC. Attempting fallback\n"));
+
+    //
+    // Fallback to the CORE_ROOT path
+    //
+
+    DWORD pathLen = GetEnvironmentVariableW(W("CORE_ROOT"), nullptr, 0);
```

Comment:
CoreCLR should not be referencing `CORE_ROOT` environment variable. This environment variable is specific to CoreRun.

Reply:
@jkotas This environment variable is purely because of a fallback scenario for testing. Would you suggest another environment variable? I agree that `CORE_ROOT` is used in CoreRun, but it is actually a relatively good name for the location of `coreclr.dll` and is what was done for the `coredistools.dll` scenario - https://github.com/dotnet/coreclr/pull/21261.

Reply:
If you have a better suggestion for the name of environment variable I will update the `coredistools` logic as well.

Reply:
We should not need environment variable to specify location of CoreCLR.dll. CoreCLR.dll can find it on its own and it does so in multiple places already. e.g. to compute System.Private.CoreLib path.

Reply:
@jkotas So much better. Thanks! 👍